### PR TITLE
refactor(payroll): replace "date_embauche" with"hiring_date"

### LINKED ISSUE #7821: Rename employee "date_embauche" to "hiring_date", for better English consistency
The French `date_embauche` field means `hiring_date` in English.  We should update this field to be in English to match the rest of our fields.

### PR DESCRIPTION
**Pull request by jniles**
Replaces the `date_embauche` with `hiring_date` for better English consistency.

Closes #7821.

### CODE REVIEW

File: server/models/migrations/next/migrate.sql [hunk 1]
```diff
@@ -13,3 +15,5 @@ ALTER TABLE `user` MODIFY `email` VARCHAR(150) DEFAULT NULL;
  */
 
 INSERT INTO `account_reference_type` (`id`, `label`, `fixed`) VALUES (8, 'FORM.LABELS.BUDGET_ANALYSIS', 1);
+
```
Comment (jmcameron) [inline server/models/migrations/next/migrate.sql:16]: Please add a comment before this line indicating the author, date, and description. Thanks.

**Review by jmcameron — CHANGES_REQUESTED** [review]
The code looks good.  
I made one minor suggestion to add a comment to the migration file.
All tests pass on my local machine.

File: server/models/migrations/next/migrate.sql (hunk 1 above)
Comment (jniles) [reply server/models/migrations/next/migrate.sql:16]: Will do, thanks!

Comment (jmcameron) [reply server/models/migrations/next/migrate.sql:16]: Thanks

**Review by jmcameron — APPROVED** [review]

**Pull request merged**